pkgs/collection/test/extensions_test.dart

```diff
@@ -1224,6 +1224,33 @@ void main() {
           );
         });
       });
+      group('.intersperse', () {
+        test('empty', () {
+          expect(iterable(<int>[]).intersperse(0), []);
+        });
+        test('single element', () {
+          expect(iterable([1]).intersperse(0), [1]);
+        });
+        test('two elements', () {
+          expect(iterable([1, 2]).intersperse(0), [1, 0, 2]);
+        });
+        test('multiple elements', () {
+          expect(iterable([1, 2, 3, 4]).intersperse(0), [1, 0, 2, 0, 3, 0, 4]);
+        });
+        test('with different separator types', () {
```

Comment:
Different from what?
(Not from the element type, which is the interesting case.)